### ISSUE TITLE
Use with https?

### ISSUE THREAD

**Issue by dandart** [tsyd/proxy-tamper#8]
Hi
I'd like to be able to use this with HTTPS traffic, is there any way of doing that?
Just getting blank responses at the moment.

**Comment by tsyd** [comment]
This is not supported at the moment. It would be a bit of work to support tampering with HTTPS traffic and would come with some caveats: you would have certificate warnings. Some MITM proxies support this by re-encrypting the traffic and signing it with a self-signed certificate that you would have to add to your browser's list of trusted certificates.

There's a Python app, [mitmproxy](https://mitmproxy.org/), that does support this. Perhaps you may find it useful.

**Comment by IonicaBizau** [comment]
@tsyd Having `https` support here would be amazing. Is it possible to add that?

**Comment by kathiedart** [comment]
I don't think it would involve not much more than a decryption and a re-encryption with a self signed certificate as above. I would have found this useful when I was using it - perhaps I'll have a go and make a PR.

**Comment by IonicaBizau** [comment]
@kathiedart That would be great––currently I have no idea how to do it. I was struggling with it for about 3 days now.

http://stackoverflow.com/q/40011633/1420197

**Comment by cbenitez101** [comment]
Hi guys!

I'´ve been triying temper-proxy for a while, great job!
I want to use it to inject advertises to hotspot users.  In my case I was triying to change the responses. If  I use Firefox, it changes the HTTP pages and if it's a HTTPS url, it just shows the page (no content injected).
When I try with a mobile device with chrome, it works OK when triying to get HTTP pages, but when I try to go to an HTTPS webpage, it gives me an error (ERR_EMPTY_RESPONSE).

Is there a way to just let it pass the HTTPS traffic as with Firefox?

Thanks for your time guys!

**Comment by tsyd** [comment]
@cbenitez101 You can usually configure your browser or OS to proxy just HTTP and to leave HTTPS traffic alone. This project does not support HTTPS, but another more complete HTTP proxy, [mitmproxy](https://mitmproxy.org/) does.
